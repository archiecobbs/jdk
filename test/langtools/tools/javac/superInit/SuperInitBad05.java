/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @summary Test invalid reference to "this" prior to superclass construction
 *
 * @compile/fail/ref=SuperInitBad05.out -XDrawDiagnostics SuperInitBad05.java
 */
class SuperInitBad05 {

    int x;
    int y;

    SuperInitBad05(int a) {
        this.x = a;
        this.y = x + 1;
        super();
    }
}
