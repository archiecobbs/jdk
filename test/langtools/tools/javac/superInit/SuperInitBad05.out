SuperInitBad05.java:15:18: compiler.err.cant.ref.before.ctor.called: x
1 error
